tsc fails when using functions in valitio/utils

## Summary

I got the following tsc error when using functions in `valtio/utils`.

```ts
import { proxy } from "valtio";
import { derive } from "valtio/utils";

const state = proxy({ count: 0 });
const derived = derive(
  {
    double: (get) => get(state).count * 2,
  },
  { proxy: state }
);
```

```console
$ tsc && vite build
node_modules/valtio/vanilla/utils/devtools.d.ts:1:63 - error TS2339: Property '__REDUX_DEVTOOLS_EXTENSION__' does not exist on type 'Window & typeof globalThis'.
```

This issue occurs with `v1.10.6` (does not occur with `v1.10.5`)
I guess #713 causes this error and the root cause is the same as https://github.com/pmndrs/zustand/issues/1205..

## Link to reproduction

https://codesandbox.io/p/sandbox/modest-curran-ptlrl7

## Check List

Please do not ask questions in issues.

- [ ] I've already opened a [discussion](https://github.com/pmndrs/valtio/discussions) before opening this issue, or already discussed in other media.

Please include a minimal reproduction.

- [x] I've added a link to a [typescript playground](https://www.typescriptlang.org/play) or [codesandbox](https://codesandbox.io) with a minimal reproduction.

--- a/docs/api/utils/devtools.mdx
+++ b/docs/api/utils/devtools.mdx
@@ -44,7 +44,7 @@ subscribe(state, () => {
 
 #### Use it with TypeScript
 
-It's recommended to install and import types from`@redux-devtools/extension` to get types correctly.
+It's recommended to install and import types from `@redux-devtools/extension` to get types correctly.
 
 ```ts
 import type {} from '@redux-devtools/extension'

--- a/docs/api/utils/devtools.mdx
+++ b/docs/api/utils/devtools.mdx
@@ -41,3 +41,12 @@ subscribe(state, () => {
   const obj = snapshot(state) // A snapshot is an immutable object
 })
 ````
+
+#### Use it with TypeScript
+
+It's recommended to install and import types from`@redux-devtools/extension` to get types correctly.
+
+```ts
+import type {} from '@redux-devtools/extension'
+import { devtools } from 'valtio/utils'
+```

--- a/src/vanilla/utils/devtools.ts
+++ b/src/vanilla/utils/devtools.ts
@@ -1,76 +1,6 @@
 import { snapshot, subscribe } from '../../vanilla.ts'
 import type {} from '@redux-devtools/extension'
 
-// Copy types to avoid import type { Config } from '@redux-devtools/extension'
-// https://github.com/pmndrs/valtio/issues/776
-type Action<T = any> = {
-  type: T
-}
-type ActionCreator<A, P extends any[] = any[]> = {
-  (...args: P): A
-}
-type EnhancerOptions = {
-  name?: string
-  actionCreators?:
-    | ActionCreator<any>[]
-    | {
-        [key: string]: ActionCreator<any>
-      }
-  latency?: number
-  maxAge?: number
-  serialize?:
-    | boolean
-    | {
-        options?:
-          | undefined
-          | boolean
-          | {
-              date?: true
-              regex?: true
-              undefined?: true
-              error?: true
-              symbol?: true
-              map?: true
-              set?: true
-              function?: true | ((fn: (...args: any[]) => any) => string)
-            }
-        replacer?: (key: string, value: unknown) => any
-        reviver?: (key: string, value: unknown) => any
-        immutable?: any
-        refs?: any
-      }
-  actionSanitizer?: <A extends Action>(action: A, id: number) => A
-  stateSanitizer?: <S>(state: S, index: number) => S
-  actionsBlacklist?: string | string[]
-  actionsWhitelist?: string | string[]
-  actionsDenylist?: string | string[]
-  actionsAllowlist?: string | string[]
-  predicate?: <S, A extends Action>(state: S, action: A) => boolean
-  shouldRecordChanges?: boolean
-  pauseActionType?: string
-  autoPause?: boolean
-  shouldStartLocked?: boolean
-  shouldHotReload?: boolean
-  shouldCatchErrors?: boolean
-  features?: {
-    pause?: boolean
-    lock?: boolean
-    persist?: boolean
-    export?: boolean | 'custom'
-    import?: boolean | 'custom'
-    jump?: boolean
-    skip?: boolean
-    reorder?: boolean
-    dispatch?: boolean
-    test?: boolean
-  }
-  trace?: boolean | (<A extends Action>(action: A) => string)
-  traceLimit?: number
-}
-type Config = EnhancerOptions & {
-  type?: string
-}
-
 // FIXME https://github.com/reduxjs/redux-devtools/issues/1097
 type Message = {
   type: string
@@ -79,6 +9,12 @@ type Message = {
 }
 
 const DEVTOOLS = Symbol()
+
+type Config = Parameters<
+  (Window extends { __REDUX_DEVTOOLS_EXTENSION__?: infer T }
+    ? T
+    : any)['connect']
+>[0]
 
 type Options = {
   enabled?: boolean

--- a/src/vanilla/utils/devtools.ts
+++ b/src/vanilla/utils/devtools.ts
@@ -1,6 +1,76 @@
 import { snapshot, subscribe } from '../../vanilla.ts'
 import type {} from '@redux-devtools/extension'
 
+// Copy types to avoid import type { Config } from '@redux-devtools/extension'
+// https://github.com/pmndrs/valtio/issues/776
+type Action<T = any> = {
+  type: T
+}
+type ActionCreator<A, P extends any[] = any[]> = {
+  (...args: P): A
+}
+type EnhancerOptions = {
+  name?: string
+  actionCreators?:
+    | ActionCreator<any>[]
+    | {
+        [key: string]: ActionCreator<any>
+      }
+  latency?: number
+  maxAge?: number
+  serialize?:
+    | boolean
+    | {
+        options?:
+          | undefined
+          | boolean
+          | {
+              date?: true
+              regex?: true
+              undefined?: true
+              error?: true
+              symbol?: true
+              map?: true
+              set?: true
+              function?: true | ((fn: (...args: any[]) => any) => string)
+            }
+        replacer?: (key: string, value: unknown) => any
+        reviver?: (key: string, value: unknown) => any
+        immutable?: any
+        refs?: any
+      }
+  actionSanitizer?: <A extends Action>(action: A, id: number) => A
+  stateSanitizer?: <S>(state: S, index: number) => S
+  actionsBlacklist?: string | string[]
+  actionsWhitelist?: string | string[]
+  actionsDenylist?: string | string[]
+  actionsAllowlist?: string | string[]
+  predicate?: <S, A extends Action>(state: S, action: A) => boolean
+  shouldRecordChanges?: boolean
+  pauseActionType?: string
+  autoPause?: boolean
+  shouldStartLocked?: boolean
+  shouldHotReload?: boolean
+  shouldCatchErrors?: boolean
+  features?: {
+    pause?: boolean
+    lock?: boolean
+    persist?: boolean
+    export?: boolean | 'custom'
+    import?: boolean | 'custom'
+    jump?: boolean
+    skip?: boolean
+    reorder?: boolean
+    dispatch?: boolean
+    test?: boolean
+  }
+  trace?: boolean | (<A extends Action>(action: A) => string)
+  traceLimit?: number
+}
+type Config = EnhancerOptions & {
+  type?: string
+}
+
 // FIXME https://github.com/reduxjs/redux-devtools/issues/1097
 type Message = {
   type: string
@@ -10,14 +80,10 @@ type Message = {
 
 const DEVTOOLS = Symbol()
 
-type EnhancerOptions = Parameters<
-  NonNullable<(typeof window)['__REDUX_DEVTOOLS_EXTENSION__']>['connect']
->[0]
-
 type Options = {
   enabled?: boolean
   name?: string
-} & EnhancerOptions
+} & Config
 
 export function devtools<T extends object>(
   proxyObject: T,

